### PR TITLE
[RNMobile] Adding WPMediaLibrary source to File block

### PR DESCRIPTION
**Pull request by etoledom**
## Description

`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2805
`WPiOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/15291
`WPAndroid`: https://github.com/wordpress-mobile/WordPress-Android/pull/13413

This PR adds the WordPress Media Library as a media source for File block.

## How has this been tested?

- Run the example app.
- Add a file block.
- Press on `Choose a file`.
  - Check that the option `WordPress Media Library` is present.
- Select such option.
  - Check that `WordPress.zip` file has been added to the block.
  - (This is mocked on the example app).

## Screenshots 

Screenshot 2020-11-18 at 11 05 18

<kbd></kbd>

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by etoledom** [comment]
@Guarani could you please take a look at the iOS side changes?
Thanks! 🙏

File: packages/react-native-editor/ios/GutenbergDemo/GutenbergViewController.swift [hunk 1]
```diff
@@ -311,12 +313,13 @@ extension GutenbergViewController: GutenbergBridgeDataSource {
...
     func gutenbergMediaSources() -> [Gutenberg.MediaSource] {
-        return [.filesApp]
+        return [.filesApp, .otherApp]
     }
 }
 
 extension Gutenberg.MediaSource {
-    static let filesApp = Gutenberg.MediaSource(id: "files-app", label: "Pick a file", types: [.image, .video, .audio, .other])
+    static let filesApp = Gutenberg.MediaSource(id: "files-app", label: "Pick a file", types: [.all])
+	static let otherApp = Gutenberg.MediaSource(id: "other-apps", label: "Other Apps", types: [.image, .video, .audio, .other])
```
Comment (guarani) [inline packages/react-native-editor/ios/GutenbergDemo/GutenbergViewController.swift:322]:
```suggestion
    static let otherApp = Gutenberg.MediaSource(id: "other-apps", label: "Other Apps", types: [.image, .video, .audio, .other])
```

File: packages/block-editor/src/components/index.native.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export {
 	default as MediaUpload,
 	MEDIA_TYPE_IMAGE,
 	MEDIA_TYPE_VIDEO,
+	MEDIA_TYPE_ALL,
```
Comment (guarani) [inline packages/block-editor/src/components/index.native.js:29]: nitpick: `MEDIA_TYPE_ANY` seems more intuitive, but that's just my opinion.

**Review by guarani — APPROVED** [review]
Looking good from the iOS side (I didn't test Android), left some comments, but no blockers.

**Comment by etoledom** [comment]
Thank you for the review @guarani.
I have updated the PR with the requested changes.

I have also updated the Android native side (cc @jd-alexander )
These changes might break WP apps PRs, so we will need to update WPAndroid and WPiOS too with the `ALL -> ANY` change.

**Comment by jd-alexander** [comment]
Hi @etoledom I updated the media source picker's title and added an icon for the "Choose from device" option that displays an icon on Android. 

Android | iOS 
--------|-------
<kbd></kbd>  |     <kbd></kbd>

Here, we have two variations of the button label for using a file from the device. 

Should we be using. 
1. Choose from device
2. Pick a file

I opted for "Choose from device" because it's used for the other blocks. Let me know what you think. 

----------

Even though this solution works I noticed that the "Choose from device" option for the File block is below the WP media option but for the other blocks, it's the first option. Below is a clear example. 

  Android | iOS 
--------|-------
<kbd></kbd>  |     <kbd></kbd>

This a minor thing. [I looked at the code and it seems the internal sources ](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/media-upload/index.native.js#L101)eg. WP Media Library is appended to the other media options which we have provided hence the ordering. Should we modify this in any way?

**Comment by etoledom** [comment]
> Here, we have two variations of the button label for using a file from the device.

We should use `Choose from device` as specified in the design file.
I have changed the title for the iOS Demo app.
WPiOS already shows `Choose from device` 👍

**Review by jd-alexander — APPROVED** [review]
The behavior works as expected based on the tests that were conducted on both platforms. LGTM 🚢

**Pull request merged**